daily_milestone.go

```diff
@@ -1,29 +1,91 @@
 package main
 
-import "flag"
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
 
-func get_project_id(base_url string, token string, projectname string, namespace string) string {
-	headers := map[string]string{"PrivateToken": token}
-	url := "https://" + base_url + "/projects"
-	page := 1
-	for {
+// Struct to be used for milestone
+//Milestone ....
+type gitLabAPI struct {
+	ID          int        `json:"id"`
+	Iid         int        `json:"iid"`
+	ProjectID   int        `json:"project_id"`
+	Title       string     `json:"title"`
+	Description string     `json:"description"`
+	StartDate   string     `json:"start_date"`
+	DueDate     string     `json:"due_date"`
+	State       string     `json:"state"`
+	UpdatedAt   *time.Time `json:"updated_at"`
+	CreatedAt   *time.Time `json:"created_at"`
+}
 
-	}
+// Initialization of logging variable
+var logger *log.Logger
 
+// LoggerSetup Initialization
+func LoggerSetup(info io.Writer) {
+	logger = log.New(info, "INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
 }
 
-//func get_milestones(base_url string, token string, project_id string)[]string{}
-//func create_milestone_data(advance int) []string{}
-//func create_milestones(base_url string, token string, project_id string, milestones string) string{}
+// Function to get project ID from the gitLabAPI
+func getProjectID(baseURL string, Token string, Projectname string, Namespace string) string {
+	project := gitLabAPI{}
+	urls := "https://" + baseURL + "/projects"
+	page := 1
+	strPage := strconv.Itoa(page)
+	s := []string{urls, "?page=", strPage}
+	completeURL := strings.Join(s, "")
+	for {
+		client := &http.Client{}
+		req, err := http.NewRequest("GET", completeURL, nil)
+		if err != nil {
+			logger.Println(err)
+			break
+		}
+		req.Header.Add("Token", Token)
+		resp, err := client.Do(req)
+		if err != nil {
+			logger.Println(err)
+			break
+		}
+		respByte, err := ioutil.ReadAll(resp.Body)
```

Comment:
You're using `ioutil.ReadAll` correctly. [Here](https://godoc.org/io/ioutil#ReadAll) is the documentation for the function, it takes an `io.Reader` interface (which is "something" (could be anything, you don't have to care about what it really is) that can be read. 

The `ReadAll` function takes this "thing" (remember, can be anything, you just have to know that it can be read) and reads it, returning a byte array `[]byte`, which is just a string under a different form (can be casted to string via `string(myByteArray[:])`

This byte array (which is your json file) can then be passed to the `json.Unmarshall` function as you did to convert it to the gitLabAPI struct.

Another option if you're still confused by ioutil.ReadAll is to use [`json.NewDecoder`](https://golang.org/pkg/encoding/json/#NewDecoder) which can directly take an `io.Reader` interface (our `resp.Body` here) and then `Decode` it this way:
```go
project := gitLabAPI{}
resp, _ := client.Do(req)
defer resp.Body.Close()
dec := json.NewDecoder(resp.Body)
dec.Decode(project)
```

You don't have to use this, it's just another way of doing things that don't use `ioutil.ReadAll` if it confuses you

Reply:
Oh Ok thanks